src/MizarData/ActiveFunctionTimePerFrameComparatorTest.cpp

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+#include <utility>
+
+#include "MizarData/ActiveFunctionTimePerFrameComparator.h"
+#include "MizarData/BaselineOrComparison.h"
+#include "MizarData/SampledFunctionId.h"
+#include "MizarData/SamplingWithFrameTrackComparisonReport.h"
+
+namespace orbit_mizar_data {
+
+using testing::DoubleNear;
+using testing::Return;
+
+namespace {
+class MockCounts {
+ public:
+  MOCK_METHOD(double, GetExclusiveRate, (SFID), (const));
+  MOCK_METHOD(uint64_t, GetTotalCallstacks, (), (const));
+};
+
+class MockFrameTrackStats {
+ public:
+  MOCK_METHOD(double, ComputeAverageTimeNs, (), (const));
+  MOCK_METHOD(uint64_t, variance_ns, (), (const));
+  MOCK_METHOD(uint64_t, count, (), (const));
+};
+
+}  // namespace
+
+constexpr uint64_t kTotalCallstacksBaseline = 100;
+constexpr uint64_t kTotalCallstacksComparison = 200;
+
+constexpr double kRateBaseline = 0.1;
+constexpr double kRateComparison = 0.15;
+
+constexpr uint64_t kFramesCntBaseline = 1000;
+constexpr uint64_t kFramesCntComparison = 1300;
+
+constexpr double kFrametimeVarBaseline = 100;
+constexpr double kFrametimeVarComparison = 150;
+
+constexpr double kAvgFrametimeBaseline = 1000;
+constexpr double kAvgFrametimeComparison = 900;
+
+constexpr SFID kArbitrarySfid(10);
+
+constexpr double kExpectedStatistic = -0.929944;
+constexpr double kPvalue = 0.352400;
+constexpr double kTol = 1e-3;
+
+class ActiveFunctionTimePerFrameComparatorTest : public ::testing::Test {
+ public:
+  ActiveFunctionTimePerFrameComparatorTest()
+      : baseline_counts_(std::in_place_t{}),
+        comparision_counts_(std::in_place_t{}),
+        baseline_frame_track_stats_(std::in_place_t{}),
+        comparison_frame_track_stats_(std::in_place_t{}) {
+    EXPECT_CALL(*baseline_frame_track_stats_, ComputeAverageTimeNs)
+        .WillRepeatedly(Return(kAvgFrametimeBaseline));
+    EXPECT_CALL(*comparison_frame_track_stats_, ComputeAverageTimeNs)
+        .WillRepeatedly(Return(kAvgFrametimeComparison));
+
+    EXPECT_CALL(*baseline_frame_track_stats_, count).WillRepeatedly(Return(kFramesCntBaseline));
+    EXPECT_CALL(*comparison_frame_track_stats_, count).WillRepeatedly(Return(kFramesCntComparison));
+
+    EXPECT_CALL(*baseline_frame_track_stats_, variance_ns)
+        .WillRepeatedly(Return(kFrametimeVarBaseline));
+    EXPECT_CALL(*comparison_frame_track_stats_, variance_ns)
+        .WillRepeatedly(Return(kFrametimeVarComparison));
+
+    EXPECT_CALL(*baseline_counts_, GetTotalCallstacks)
+        .WillRepeatedly(Return(kTotalCallstacksBaseline));
+    EXPECT_CALL(*comparision_counts_, GetTotalCallstacks)
+        .WillRepeatedly(Return(kTotalCallstacksComparison));
```

Comment:
Do you expect all these calls in every single test? Probably yes, but you have `ON_CALL` if not.

Reply:
Yeah, the comparator needs all of it. I even suspect every "reasonable" test solving the same problem would need it all.

Reply:
But thanks for the advice.